### ISSUE TITLE
Previous Parcel Displaying on Site Location Map After Updating Site Location

### ISSUE THREAD

**Issue by rsparker-utah** [uic-utah/uic-inventory#271]
The previous parcel is displaying alongside the updated parcel on the "Site Location" page map after updating a site's location. I noticed this possible bug while going through some testing for #250. 

These are the steps I took to encounter this bug ([site 72](https://uic-inventory.dev.utah.gov/site/72/add-location) used for this example):

1.) I navigated to the "Site Location" page: 
![image](https://github.com/uic-utah/uic-inventory/assets/49735267/172a573e-26df-4555-bc9a-7a37344bf80b)
2.) Clicked the "Select Parcels" tool to clear the existing parcel selection: ![image](https://github.com/uic-utah/uic-inventory/assets/49735267/2d926af2-d123-4a7d-9024-abd4680f8f73)
3.) Selected a new parcel: 
![image](https://github.com/uic-utah/uic-inventory/assets/49735267/12dd60c0-ad3e-443f-8492-33bb4ab8cbce)
4.) Hit "Next" to save: 
![image](https://github.com/uic-utah/uic-inventory/assets/49735267/b28d1163-19fe-45da-a6d2-33d8b541c9ab)
5.)  Navigated back to the "Site Location" page and both the old and new parcels were being displayed on the map: 
![image](https://github.com/uic-utah/uic-inventory/assets/49735267/4e195e61-edc9-485f-a75b-25a70c1a0bbf)
6.) If I refreshed the page, the old parcel would disappear from the map:
![image](https://github.com/uic-utah/uic-inventory/assets/49735267/27a70626-258d-4e0a-8577-894b294c6150)

**Comment by steveoh** [comment]
I believe I have this fixed. Please test all the scenarios you can for this screen.

**Comment by rsparker-utah** [comment]
I ran through some testing and the parcel select/deselect is working perfectly. The bug with the previous parcel displaying alongside the updated parcel has been resolved. The only other thing I came across during my testing was that I was able to edit the parcel location for a site (i.e., site [82](https://uic-inventory.dev.utah.gov/site/84/add-location)) that had submitted inventories as a public user. According to issue #143 this is still pending, so this is more of a reminder note for myself rather than an issue.
